src/app/shared/services/share/share.service.ts

```diff
@@ -0,0 +1,53 @@
+import { Injectable } from "@angular/core";
+import { SyncServiceBase } from "../syncService.base";
+import { Share, ShareOptions } from "@capacitor/share";
+import { TemplateActionRegistry } from "../../components/template/services/instance/template-action.registry";
+
+@Injectable({
+  providedIn: "root",
+})
+export class ShareService extends SyncServiceBase {
+  constructor(private templateActionRegistry: TemplateActionRegistry) {
+    super("Share");
+    this.initialise();
+  }
+
+  private initialise() {
+    this.registerTemplateActionHandlers();
+  }
+
+  private registerTemplateActionHandlers() {
+    this.templateActionRegistry.register({
+      share: async ({ args }) => {
+        const [actionId, ...shareArgs] = args;
+        const childActions = {
+          text: async () => {
```

Comment:
nit(non-blocking)
Is there a reason to split the text/url functionality? It might make more sense to just keep as a single `share` action and allow parameters the exact same as the capacitor share api (and underlying web-share api), as this simplifies the process of documentation and supporting any future features

Reply:
That sounds sensible, but I don't think we have a way to support key/value parameters being passed to an action currently. We interpret the arguments as an ordered list – I think trying to pass in key/value pairs might reach the limits of our list separator syntax, but I could well be wrong. Maybe something to be considered alongside working towards a universal syntax between the app and chatbot authoring, I have an RFC on the back burner.

Reply:
From my authoring point of view it would make sense if `share` just expects a string (which could be text or URL) - but I appreciate I don't know the full picture

Reply:
I think every action can decide how to process their own args, so for example we could read the share action arg and process depending on whether authored as a single string or as a parameter list
```
click | share | url: https://some-url.com; text: Here is my text;
```
We would just need to parse the second arg as a param list

Whereas
```
click | share | https://some-url.com
```
could imply a default of `url: https://some-url.com` and still remain functional.

I think there should already be a utility function somewhere to parse string parameter lists, although might need a little extra testing to see how best to handle case where only a string included